bundles/org.openhab.ui/web/src/js/openhab/sse.js

```diff
@@ -17,61 +18,74 @@ function newSSEConnection (path, readyCallback, messageCallback, errorCallback,
     const creds = getBasicCredentials()
     headers['Authorization'] = 'Basic ' + btoa(creds.id + ':' + creds.password)
   }
-  if (Object.keys(headers).length > 0) {
-    eventSource = new EventSourcePolyfill(path, { headers })
-  } else {
-    eventSource = new NativeEventSource(path)
-  }
-
-  eventSource.addEventListener('ready', (e) => {
-    readyCallback(e.data)
-  })
 
-  eventSource.addEventListener('alive', (e) => {
-    let evt = JSON.parse(e.data)
-    eventSource.setKeepalive(evt.interval)
-    if (heartbeatCallback) {
-      heartbeatCallback(true)
+  function initEventSource () {
+    if (Object.keys(headers).length > 0) {
+      eventSource = new EventSourcePolyfill(path, { headers })
+    } else {
+      eventSource = new NativeEventSource(path)
     }
-  })
 
-  eventSource.onmessage = (event) => {
-    let evt = JSON.parse(event.data)
-    messageCallback(evt)
-  }
+    eventSource.addEventListener('ready', (e) => {
+      readyCallback(e.data)
+    })
 
-  eventSource.onopen = (event) => {
-  }
+    eventSource.addEventListener('alive', (e) => {
+      let evt = JSON.parse(e.data)
+      eventSource.setKeepalive(evt.interval)
+      if (heartbeatCallback) {
+        heartbeatCallback(true)
+      }
+    })
 
-  eventSource.onerror = () => {
-    console.warn('SSE error')
-    eventSource.clearKeepalive()
-    if (errorCallback) {
-      errorCallback()
+    eventSource.onmessage = (event) => {
+      let evt = JSON.parse(event.data)
+      messageCallback(evt)
     }
-    if (eventSource.readyState === 2) {
-      console.log('%c=!= Event source connection broken...', 'background-color: red; color: white')
+
+    eventSource.onopen = (event) => {
+      reconnectSeconds = 1
     }
-  }
 
-  eventSource.setKeepalive = (seconds = 10) => {
-    console.debug('Setting keepalive interval seconds', seconds)
-    eventSource.clearKeepalive()
-    eventSource.keepaliveTimer = setTimeout(() => {
-      console.warn('SSE timeout error')
-      if (heartbeatCallback) {
-        heartbeatCallback(false)
```

Comment:
Looks like `heartbeatCallback(false)` is now not called at all, should this be moved into the `onerror` function below?

Reply:
I just realized that I had the wrong train of thought when I removed it.
I will revert this change.

I don’t think it should be moved to onerror. AFAIK onerror is triggered on something different then heartbeat timeout.

Reply:
@digitaldan I have reverted that change, can you please have another look and approve if everything is okay?